tests/helpers/perf_run_helper.rb

```diff
@@ -944,75 +945,253 @@ def get_process_hash(perf_result_processes)
     baseline = get_baseline_result
-    baseline&.each do |key, value|
-      # orchestration has a jump in memory usage in 2019.2 due to plan executor
-      # https://tickets.puppetlabs.com/browse/BOLT-986
-      # Currently we don't have a way to whitelist that... so just turning off the check
-      # Ticket to make a way https://tickets.puppetlabs.com/browse/SLV-515
-      next if key.to_s == "process_orchestration_services_release_avg_mem" && BEAKER_PE_VER =~ /^2019.2*/
-
-      assert_value = if key.to_s == "avg_response_time"
-                       gatling_result.avg_response_time.to_f
-                     elsif key.to_s.start_with? "process"
-                       process_results[key.to_s].to_f
-                     else
-                       atop_result.send(key).to_f
-                     end
-
-      puts "Value for #{key} is baseline: #{value} Actual: #{assert_value}"
-      next if key.to_s.start_with?("process") && key.to_s.end_with?("cpu")
-
-      max_baseline_variance = \
-        if key.to_s == "process_orchestration_services_release_avg_mem"
-          15
-        else
-          10
-        end
-      next unless value.is_a? Integer
+    delta_data = baseline_to_results_delta(baseline, atop_result, gatling_result)
+    assert_later(validate_baseline_delta(delta_data), "Things are really fucked up now")
+  end
 
-      assert_later((assert_value - value) / value * 100 <= max_baseline_variance,
-                   "The value of #{key} '#{assert_value}' " \
-                   "was not within #{max_baseline_variance}% " \
-                   "of the baseline '#{value}'")
+  # rubocop: enable Naming/AccessorMethodName
+
+  # Calculate the variances on the data set provided and return a hash of the
+  # failing elements with the variance calculation appended to each key.
+  #
+  # @param  [Hash] data  Data to validate { key: [v0, v2], ... }
+  #
+  # @return [Hash]  Hash of failing performance keys associated with there values and variance
+  def find_failing_variances(data)
+    deltas = data.transform_values { |v| v[1].to_f / v[0] }
+    failures = deltas.select { |_k, v| (1 - v).abs > MAX_BASELINE_VARIANCE }
+    # exception for PROC_ORCH_REL_MEM
+    if failures.include? PROC_ORCH_REL_MEM
+      failures.delete(PROC_ORCH_REL_MEM) if (1 - failures[PROC_ORCH_REL_MEM]).abs < MAX_BASELINE_VARIANCE_ORCH_REL_MEM
     end
+    failures.each_with_object({}) { |(k, v), hash| hash[k] = data[k] + [v] }
+  end
+
+  # Determine if the provided results data set contains any failures.
+  #
+  # @param  [Hash] data  Data to validate { key: [v0, v2], ... }
+  #
+  # @return [Boolean]
+  def validate_baseline_delta(data)
+    failures = find_failing_variances(data)
+    return true if failures.empty?
+
+    failures.each do |k, v|
+      variance = ((v.last - 1) * 100).round(2)
+      logger.error("Result '#{k}' is outside tolerances: baseline: #{v.first}; result: #{v[1]}; variance: #{variance}%")
```

Comment:
mixing v.first and v[1] seems odd.  woudl have expected v[0] and v[1] for consistency.

Reply:
fixed.